src/Context/Auth.context.js

```diff
@@ -0,0 +1,86 @@
+import axios from "axios";
+import { createContext, useContext, useState, useEffect } from "react";
+import { useReducerContext } from "./Reducer.context";
+
+const AuthContext = createContext();
+
+const AuthProvider = ({ children }) => {
+  const { dispatch } = useReducerContext();
+  const encodedToken = localStorage.getItem("StormKeepToken");
+  const [userState, setUserState] = useState([]);
+
+  const login = async (userDetails) => {
+    try {
+      const { data } = await axios.post(`/api/auth/login`, {
+        email: userDetails.email,
+        password: userDetails.password,
+      });
+      // saving the encodedToken in the localStorage
+      localStorage.setItem("StormKeepToken", data.encodedToken);
+      localStorage.setItem("StormKeepUser", data.foundUser.firstName);
+      dispatch({ type: "SUCCESS_TOAST", payload: "Log In Successful" });
+    } catch (error) {
+      dispatch({ type: "ERROR_TOAST", payload: error.response.data.errors });
+    }
+  };
+
+  const signup = async (userDetails) => {
+    try {
+      const { data } = await axios.post(`/api/auth/signup`, {
+        firstName: userDetails.user,
+        email: userDetails.email,
+        password: userDetails.passwordOne,
+      });
+      // saving the encodedToken in the localStorage
+      localStorage.setItem("StormKeepToken", data.encodedToken);
+      localStorage.setItem("StormKeepUser", data.createdUser.firstName);
+      dispatch({ type: "SUCCESS_TOAST", payload: "Sign Up Successful" });
+    } catch (error) {
+      console.log(error);
```

Comment:
I think it can also shown on toast? Right?

Reply:
Thanks for pointing it out i actually forgot to show error with a toast here